apps/prairielearn/src/sync/fromDisk/courseInstances.ts

```diff
@@ -13,10 +13,30 @@ import * as infofile from '../infofile.js';
 
 const sql = sqldb.loadSqlEquiv(import.meta.filename);
 
-export function generateEnrollmentCode() {
-  /** A 12-character hex string should be resistant to brute force attacks. These do not have to be unique. */
-  // Similar to https://github.com/PrairieLearnInc/PrairieTest/blob/25228ee37c60b51d7d3b38240dcafa5d44bb2236/src/models/courses.ts#L526-L527
-  return randomBytes(6).toString('hex');
+export async function uniqueEnrollmentCode() {
+  let enrollmentCode = generateEnrollmentCode();
+  while (
```

Comment:
Nit, I often find a `while (true)` loop easier to read:

```ts
while (true) {
  const enrollmentCode = generateEnrollmentCode();
  if (await selectExistingEnrollmentCode() === null) {
    return enrollmentCode;
  }
}
```

Reply:
Done in 3a7987400b55eaf9390f99eab18a94f0c3201bcd.